lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc

```diff
@@ -173,15 +173,32 @@ function dosomething_northstar_perform_authorization_actions() {
  * @param $account
  */
 function dosomething_northstar_user_logout($account) {
-  $is_openid_login = ! empty($_SESSION['DOSOMETHING_NORTHSTAR_OPENID_CONNECT']);
-  $northstar_logout_url = url(NORTHSTAR_URL . '/logout', [
-    'query' => ['redirect' => url('/', ['absolute' => TRUE])],
-  ]);
+  $user = user_load($account->uid);
 
   // Destroy the current session, and reset $user to the anonymous user.
   session_destroy();
 
-  drupal_goto($is_openid_login ? $northstar_logout_url : '/');
+  $is_openid_login = ! empty($user->field_refresh_token[LANGUAGE_NONE][0]['value']);
```

Comment:
I decided against using `dosomething_user_get_field` here because it logs to Watchdog if the field isn't set and that seemed like it would create a lot of noise.